src/util_test.cc

```diff
@@ -357,7 +446,7 @@ TEST(CanonicalizePath, NotNullTerminated) {
   len = strlen("foo/../file");
   CanonicalizePath(&path[0], &len, &unused);
   EXPECT_EQ(strlen("file"), len);
-  EXPECT_EQ("file ./file bar/.", string(path));
+  EXPECT_EQ("file../file bar/.", string(path));
```

Comment:
Hm ... should the unit test really care about the rest of the string?

Reply:
Yes, the point is to verify that this does not touch the rest of the string, including from the input (hence the caller should add a terminating zero if it needs it). I've added a comment to clarify.